src/storage/v3/shard_rsm.cpp

```diff
@@ -322,9 +331,14 @@ std::optional<msgs::ExpandOneResultRow> GetExpandOneResult(
   if (!source_vertex) {
     return std::nullopt;
   }
-
+  std::optional<std::map<PropertyId, Value>> src_vertex_properties;
   /// Fill up source vertex properties
-  auto src_vertex_properties = FillUpSourceVertexProperties(v_acc, req);
+  if (req.src_vertex_properties) {
+    src_vertex_properties = FillUpSourceVertexProperties(v_acc, req);
+  } else {
+    src_vertex_properties = CollectAllPropertiesFromAccessor(*v_acc, storage::v3::View::NEW, schema);
+  }
```

Comment:
Please rethink this logic, because the `FillUpSourceVertexProperties` does the same check. Maybe try to unify the logic in a better way.

Reply:
The handlers are (slightly) a spagheti mess. As m1 is the priority, I won't spend much on this side at all. I will see if I can remove 1-2 lines but I won't pay much attention to it.